File: 860_Lemonade_Change.js
var lemonadeChange = function(bills) {
    let bill5 = 0;
    let bill10 = 0;
    for(let i = 0; i < bills.length; i++){
        if(bills[i] == 5) bill5++;
        else if(bills[i] == 10)
        {
            bill5--;
            bill10++;
        } else {
            if(bill10 > 0)
            {
                bill10--;
                bill5--;
            }
            else if(bill5 < 3) return false;
            else bill5 -= 3;
        }
        if(bill5 < 0 || bill10 < 0) return false;
    }
    return true;
};

File: 860_Lemonade_Change.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
       int bill5 = 0, bill10 = 0, bill20 = 0, i = 0;
       while(i < bills.size()){
        if(bills[i] == 5) bill5++;
        else if(bills[i] == 10){
            bill10++;
            if(bill5 < 1) return false;
            bill5--;
        } else {
            bill20++;
            if(bill10 && bill5){
                bill10--;
                bill5--;
            } else if(bill5 >= 3) bill5 -= 3;
            else return false;
        }
        i++;
       }
        return true;
    }
};
